vertx-web-client/src/main/java/io/vertx/ext/web/client/impl/cache/CachedHttpResponse.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2021 Red Hat, Inc.
+ *
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *  The Eclipse Public License is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  The Apache License v2.0 is available at
+ *  http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ */
+package io.vertx.ext.web.client.impl.cache;
+
+import io.vertx.core.MultiMap;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpVersion;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.client.impl.HttpResponseImpl;
+import io.vertx.ext.web.client.spi.CacheStore;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.Serializable;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Collections;
+
+/**
+ * A serializable object to be stored by a {@link CacheStore}.
+ *
+ * @author <a href="mailto:craigday3@gmail.com">Craig Day</a>
+ */
+public class CachedHttpResponse implements Serializable {
```

Comment:
can we avoid using `Serializable` and let the actual cache deal with serialization when needed

Reply:
Hm, why do we want to avoid it? How would a cache store know which attributes to serialize to ensure that the cached response they return will work?

Reply:
cache store should have access to the various field of the CachedHttpResponse and be responsible for the serialization.